141617729 persisting food jc

jdconrad89 opened the pull request:
@DavidKnott @wlffann @case-eee. I've completed card number 141617729 for Persisting Food. I added localStorage in two places. In the add food function it is set up as
 ```
localStorage.setItem(name, foodRow);
``` 
 (if you have questions about what this is I can answer them). Also outside of all of the functions I put 
```   
for (var i = 0; i < localStorage.length; i++){
    $("tbody").append(localStorage.getItem(localStorage.key(i)));
    add_delete_functionality();
  }
```
This is happening everytime we reload the page it will cycle through everything that is in localStorage and display it on the page. You will notice that up here is has append vs prepend. This is done to ensure that the chart will always look the same when you reload it (the way it cycles through the localStorage)

jdconrad89 committed:
Implement localStorage to allow us to have persisting items

jdconrad89 committed:
141617729 deletes unneeded functionality

As I was writing the PR comment I realized I didn’t need the clear text
function when I was reloading the page with the persisted Items

case-eee commented on lib/foods.js:
```diff
@@ -1,65 +1,75 @@
 $('document').ready(function() {
+  for (var i = 0; i < localStorage.length; i++){
```
since this doesn't rely on anything specific within the document ready function, i'd suggest breaking this out into its own function 👍

jdconrad89 replied:
@case-eee just want to double check to make sure I completely understand localStorage. In a way it functions like a session in a Rails App so I can call it anywhere and we will have access to what is in it correct?

case-eee replied:
@jdconrad89 you can think of it in a way as a session for now. we'll talk about it during class tomorrow morning 🎉

case-eee commented on lib/foods.js:
```diff
@@ -1,65 +1,75 @@
-                      <td class='food-calories'>` + calories + `</td>
-                      <td class='food-delete'><button>-</button></td>
-                    </tr>`
+    <td class='food-name'>` + name + `</td>
```
same here - constructing this row could be its own function 🎉

case-eee commented on lib/foods.js:
```diff
@@ -1,65 +1,75 @@
-  $("td.food-delete button").on('click', function(){
+  function add_delete_functionality(){
+    $("td.food-delete button").on('click', function(){
       $(this).parent().parent().remove();
```
could we be more specific with determining what to remove here rather than using parent.parent? what happens if the structure of the html changes?

wlffann replied:
@DavidKnott

jdconrad89 replied:
@case-eee I had mentioned doing something like
```
$("#food-list").on("click", "button", function() {
    $(this).closest("tr").remove();
```
is this more like what you were thinking?

DavidKnott replied:
I removed the parent().parent() in my branch, thanks for spotting it @case-eee!

case-eee reviewed (commented):
@jdconrad89 good work on this 🎉 don't forget to tag @icorson3 too!

wlffann commented on lib/foods.js:
```diff
@@ -1,65 +1,75 @@
 $('document').ready(function() {
+  for (var i = 0; i < localStorage.length; i++){
+    $("tbody").append(localStorage.getItem(localStorage.key(i)));
```
This is nice! I like that it will only run when the page is loaded. Will it add the `<td>`s as well?

jdconrad89 replied:
@wlffann yes ma'am because when we add the item into local storage it is set up as  the foodRow with all the html

wlffann commented on lib/foods.js:
```diff
@@ -1,65 +1,75 @@
 
-function check_for_name(name) {
-  if (name == ""){
+  function check_for_name(name) {
```
Do these need to be wrapped in the $(document).ready()? It looks like they weren't before...

DavidKnott replied:
I don't think so because these functions are called inside the document ready.

wlffann reviewed (commented):
Looks great! I'm glad that this was relatively straight forward. Thanks for the really descriptive PR message as well!

DavidKnott commented:
It looks good to me and a lot of the changes that @case-eee suggested I'm making.  Do you guys feel okay merging it?

jdconrad89 committed:
141617729 create the loadTheFoodItem function

Pull
```
for (var i = 0; i < localStorage.length; i++){
    $("tbody").append(localStorage.getItem(localStorage.key(i)));
    add_delete_functionality();
  }
```
out into its own function

jdconrad89 commented:
@DavidKnott @case-eee @wlffann @icorson3 Pulled the 
```
for (var i = 0; i < localStorage.length; i++){
    $("tbody").append(localStorage.getItem(localStorage.key(i)));
    add_delete_functionality();
  }
```
out into a function. In order to make it work i still had to call the function within
```$(document).ready()```
as I understand it that everything inside that will load the first time. I think this is what you meant @case-eee?

case-eee commented:
@jdconrad89 that's exactly what I meant. thanks for clarifying!
